runtime/src/common/quantity.rs

```diff
@@ -47,6 +49,14 @@ impl From<u64> for Quantity {
     }
 }
 
+impl TryFrom<&Quantity> for u64 {
+    type Error = anyhow::Error;
```

Comment:
Would be better to use a more specific error, maybe one of the existing ones could be reused?